src/offline.resources.ts

```diff
@@ -0,0 +1,141 @@
+  const metadataUrl = '/ws/rest/v1/metadatamapping/termmapping?v=full&code=emr.primaryIdentifierType';
+  await cacheUrl(metadataUrl);
+  const primaryIdentifierTypeResponse = await openmrsFetch(metadataUrl, { signal: abortController?.signal });
+
+  const identifierUrl = `/ws/rest/v1/patientidentifiertype/${primaryIdentifierTypeResponse.data.results[0].metadataUuid}`;
+  await cacheUrl(identifierUrl);
+  const { data: metadata } = await openmrsFetch(identifierUrl, { signal: abortController?.signal });
+
+  return {
+    name: metadata.name,
+    fieldName: camelCase(metadata.name),
+    required: metadata.required,
+    uuid: metadata.uuid,
+    format: metadata.format,
+    isPrimary: true,
+  };
+}
+
+async function fetchSecondaryIdentifierTypes(
+  abortController?: AbortController,
+): Promise<Array<FetchedPatientIdentifierType>> {
+  const metadataUrl = '/ws/rest/v1/metadatamapping/termmapping?v=full&code=emr.extraPatientIdentifierTypes';
+  await cacheUrl(metadataUrl);
+  const secondaryIdentifierTypeResponse = await openmrsFetch(metadataUrl, { signal: abortController?.signal });
+
+  if (secondaryIdentifierTypeResponse.data.results) {
+    const extraIdentifierTypesSetUuid = secondaryIdentifierTypeResponse.data.results[0].metadataUuid;
+    const memberMetadataUrl = `/ws/rest/v1/metadatamapping/metadataset/${extraIdentifierTypesSetUuid}/members`;
+    await cacheUrl(memberMetadataUrl);
+    const metadataResponse = await openmrsFetch(memberMetadataUrl, { signal: abortController?.signal });
+
+    if (metadataResponse.data.results) {
+      return await Promise.all(
+        metadataResponse.data.results.map(async setMember => {
+          const identifierUrl = '/ws/rest/v1/patientidentifiertype/' + setMember.metadataUuid;
+          await cacheUrl(identifierUrl);
+          const type = await openmrsFetch(identifierUrl, { signal: abortController?.signal });
+
+          return {
+            name: type.data.name,
+            fieldName: camelCase(type.data.name),
+            required: type.data.required,
+            uuid: type.data.uuid,
+            format: secondaryIdentifierTypeResponse.data.format,
+            isPrimary: false,
+          };
+        }),
+      );
+    }
+  }
+
+  return [];
+}
+
+async function fetchIdentifierSources(identifierType: string, abortController?: AbortController) {
+  const url = '/ws/rest/v1/idgen/identifiersource?v=full&identifierType=' + identifierType;
+  await cacheUrl(url);
+  return await openmrsFetch(url, { signal: abortController?.signal });
+}
+
+function fetchAutoGenerationOptions(identifierType: string, abortController?: AbortController) {
+  // return openmrsFetch('/ws/rest/v1/idgen/autogenerationoption?v=full&identifierType=' + identifierType, {
+  //   signal: abortController.signal,
+  // });‚
+  return Promise.resolve(mockAutoGenerationOptionsResult);
+}
+
+async function cacheUrl(url: string) {
```

Comment:
Something like this could prob be moved to `esm-offline` at some point. I'd like to postpone it for the moment though to see it stabilize with respect to other repos.

Also thinking about extending the `openmrsFetch` options with sth like `{ cacheUrl: boolean }`...

Reply:
Alright - good idea!